tests/phpunit/tests/media.php

```diff
@@ -3853,6 +3853,48 @@ public function test_wp_filter_content_tags_does_not_lazy_load_images_in_header(
 		$this->assertSame( '<div class="wp-site-blocks">' . $expected_template_content . '</div>', $html );
 	}
 
+	/**
```

Comment:
It might be worth adding a unit test that just tests `wp_get_loading_attr_default()` with special contexts. I assume that pre-existing tests cover non-special (ordinary? 😄) contexts. The special contexts data provider can then be expanded later and may be useful in other tests.

```php
/**
 * Tests that wp_get_loading_attr_default() returns false for special contexts.
 *
 * @ticket 58089
 *
 * @covers ::wp_get_loading_attr_default
 *
 * @dataProvider data_special_contexts
 *
 * @param string $context Context for the element for which the `loading` attribute value is requested.
 */
public function test_wp_get_loading_attr_default_should_return_false_for_special_contexts( $context ) {
    $this->assertFalse( wp_get_loading_attr_default( $context ) );
}

/**
 * Data provider.
 *
 * @return array[]
 */
public function data_special_contexts() {
    return array(
        'the_post_thumbnail'      => array( 'context' => 'the_post_thumbnail' ),
        'wp_get_attachment_image' => array( 'context' => 'wp_get_attachment_image' ),
}
```

Reply:
@costdev Great idea! Note however that it should only return `false` when those contexts are used within the `the_content` filter, so I've updated your code a bit to make it two tests, one for each situation, both using the same data provider. See https://github.com/WordPress/wordpress-develop/pull/4300/commits/f5ffa3fb0e5781f81f30971f608a401d25f1e96b

Reply:
Good point, I forgot about `the_content` filter!